tests/foundry/lemmas.k

```diff
@@ -0,0 +1,113 @@
+requires "../contracts.k"
+
+module TEST-HELPER-LEMMAS
+  imports ASSERTTEST-CONTRACT
+
+  syntax Bytes ::= testFunction() [function]
+  syntax Set ::= testFunction2(Bytes) [function]
+
+  rule testFunction() => AssertTest.test_assert_true() [simplification]
+  rule testFunction() => #binRuntime(AssertTest) [simplification]
+  rule testFunction2(PGM) => #computeValidJumpDests(PGM) [simplification]
+
+endmodule
+
+module SUM-TO-N-INVARIANT
+
+  imports LOOPSTEST-CONTRACT
+
+    rule N xorInt maxUInt256 <Int S => false
+    requires #rangeUInt(256, S +Int ((N *Int (N +Int 1)) /Int 2))
+    andBool #rangeUInt(256, S)
+    andBool #rangeUInt(256, N)
+    [simplification]
```

Comment:
It should be sufficient to have 
```
andBool 0 <=Int S andBool 0 <=Int N
``` 
instead of the `#rangeUInt`s.

This is really `maxUInt256 <Int S +Int N => false`, isn't it? I wonder how the `xorInt` gets introduced...

How about having just
```
rule N xorInt maxUInt256 => maxUInt256 -Int N 
  requires #rangeUInt(256, N)
  [simplification]
```
then this lemma will become provable given that the lemma below with `chop(S +Int N)` is provable (I've tried it with KEVM master).

Reply:
`X xorInt maxUInt256` is introduced as an overflow check for addition by the Solidity compiler. Observe (we're trying to compute `S +Word N`):

- `S +Word N` overflows if `pow256 <=Int S +Int N`.
- ... if `pow256 -Int N <=Int S`.
- `pow256 -Int N == notInt N == maxUInt256 xorInt N` if `#rangeUInt(256, N)`.
- `S +Word N` overflows if `N xorInt maxUInt256 <=Int S`.

So it's a 1-operator way to check "will the addition overflow?"

Reply:
Both lemmas work, but I replaced the existing one with the one @PetarMax suggested because it is simpler.